src/mqt/bench/devices/device.py

```diff
@@ -0,0 +1,198 @@
+
+    def get_two_qubit_gates(self) -> set[str]:
+        """
+        Get the set of two-qubit gates supported by the device.
+
+        Returns
+        list of two-qubit gates
+        """
+        assert self.calibration is not None
+        return {
+            gate
+            for qubit1, qubit2 in self.coupling_map
+            for gate in self.calibration.two_qubit_gate_fidelity[(qubit1, qubit2)]
+        }
+
+    def sanitize_device(self) -> None:
+        """
+        Tries to sanitize the device information so that it produces the least amount of problems when used.
+
+        It is assumed that any edge, where the average two-qubit gate fidelity is 0, is not a valid edge.
+        Thus, such edges are removed from the coupling map.
+
+        It is ensured that
+         * all single-qubit gates have fidelity data for all qubits in the device
+         * all two-qubit gates have fidelity data for all qubit pairs in the coupling map.
+        This is accomplished by substituting the missing fidelity data with the average fidelity for the gate.
+        """
+        if self.calibration is None:
+            return
+
+        # ensure that all single-qubit gates have fidelity data for all qubits in the coupling map
+        for gate in self.get_single_qubit_gates():
+            avg_fidelity = self.calibration.compute_average_single_qubit_gate_fidelity(gate)
+            for qubit in range(self.num_qubits):
+                if (
+                    gate not in self.calibration.single_qubit_gate_fidelity[qubit]
+                    or self.calibration.single_qubit_gate_fidelity[qubit][gate] == 0
+                ):
+                    self.calibration.single_qubit_gate_fidelity[qubit][gate] = avg_fidelity
+
+        # remove any edge from the coupling map that has a fidelity of 0 for all gates.
+        self.coupling_map = [
+            edge
+            for edge in self.coupling_map
+            if all(fidelity != 0 for fidelity in self.calibration.two_qubit_gate_fidelity[tuple(edge)].values())
+        ]
+
+        # ensure that all two-qubit gates have fidelity data for all edges in the coupling map
+        for gate in self.get_two_qubit_gates():
+            avg_fidelity = self.calibration.compute_average_two_qubit_gate_fidelity(gate)
+            for edge in self.coupling_map:
+                if (
+                    gate not in self.calibration.two_qubit_gate_fidelity[tuple(edge)]
+                    or self.calibration.two_qubit_gate_fidelity[tuple(edge)][gate] == 0
+                ):
+                    self.calibration.two_qubit_gate_fidelity[tuple(edge)][gate] = avg_fidelity
+
+        # remove any fidelity data for edges that are not in the coupling map
+        self.calibration.two_qubit_gate_fidelity = {
+            tuple(edge): self.calibration.two_qubit_gate_fidelity[tuple(edge)] for edge in self.coupling_map
+        }
```

Comment:
A few edges in the ibm_washington calibration actually have gate fidelities equal to one.
I assume they were removed at this point since that seems unphysical.
The removal of the `self.calibration.two_qubit_gate_fidelity` then only happens for the respective edges.

Reply:
That is IBMs way of saying this edge is supposed to be part of the device but calibration resulted in the edge not being functional. IIRC this actually happened in quite some of the larger IBM backends.

Reply:
I see. Thanks a lot for checking. Now this PR is really ready to be merged 👍